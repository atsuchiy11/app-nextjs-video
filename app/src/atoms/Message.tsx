import * as React from 'react'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'

const Message = () => {
	return (
		<>
			<Typography>message!</Typography>
			<Button onClick={() => window.alert('With typescript and Jest')}>
				button
			</Button>
		</>
	)
}
export default Message
