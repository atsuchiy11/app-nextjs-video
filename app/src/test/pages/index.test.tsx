import React from 'react'
import { render, fireEvent } from '../testUtils'
import Message from 'src/atoms/Message'

describe('Home page', () => {
	it('show message', () => {
		const messageRender = render(<Message />, {})
		expect(messageRender.getAllByText('message!')).toHaveLength(1)
	})

	it('matches snapshot', () => {
		const { asFragment } = render(<Message />, {})
		expect(asFragment()).toMatchSnapshot()
	})

	it('clicked button', () => {
		const { getByText } = render(<Message />)
		window.alert = jest.fn()
		fireEvent.click(getByText('button'))
		expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
	})
})
