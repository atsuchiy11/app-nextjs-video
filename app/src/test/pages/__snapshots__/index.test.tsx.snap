// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    message!
  </p>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
